client.go

```diff
@@ -0,0 +1,93 @@
+package manifestival
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Client interface {
+	Create(obj *unstructured.Unstructured, options *metav1.CreateOptions) error
+	Update(obj *unstructured.Unstructured, options *metav1.UpdateOptions) error
+	Delete(obj *unstructured.Unstructured, options *metav1.DeleteOptions) error
+	Get(obj *unstructured.Unstructured, options *metav1.GetOptions) (*unstructured.Unstructured, error)
+}
+
+// Functional options pattern
+type ClientOption func(*ClientOptions)
+
+type ClientOptions struct {
```

Comment:
Hmm, I see why you did this but I'm not sure I like it.

Looking at its usage below, it seems like this is used make the interface less annoying (by allowing to pass nothing as options) and to reduce the amount of distinct functions needed. This comes at the cost of allowing me to pass options to functions that will just do nothing (like GracePeriodSeconds to Get or something).

If you don't think that slight hole in the typesafety is an issue, leave it as is. 

Alternatively we could straight up pass through the `metav1` options at the cost of having slightly more verbosity in the API. The only place where we need to "proxy" them is in `Apply` where you need both create and update options but there it seems trivial to construct one from the other.

I'm not married to either way, if you read this and still prefer the implemented approach, that's fine with me!

Reply:
Yeah, this was a giant trade-off. I don't love the potential for ignored options, but I hated it less than creating a new type for every option type. And I *really* hate for clients to have to pass an empty slice in when the common case will probably never include any options.

I'm content to go with this until someone complains about it louder or a better solution presents itself.

Reply:
wfm!